### ISSUE TITLE
[Pagination] Change Select's label to "Page number"

### ISSUE THREAD

**Issue by alina-jacob** [carbon-design-system/carbon#17974]
## Acceptance criteria
- Current implementation of the pagination nav can have an improved screen reader announcement. 

<br>

**Problem:**
The screen reader announcement for the pagination component is a bit odd. It potentially announces "of 40 pages" twice. 
This is the current implementation: Page number of 40 pages, 12

**Solution:**
Implement the label in a way that the screen-reader sounds similar to this: "Page, 12, of 40 pages". 
PS: Combobox is the assigned role of the component.

- [x] Select's label for Pagination should read as "Page" instead of "Page number, of 40 pages".
- [ ] Update website doc image to reflect the same ([issue here](https://github.com/carbon-design-system/carbon-website/issues/4351))

![Image](https://github.com/user-attachments/assets/7b4f1acd-9759-4e78-b39a-afb3d7ab1209)

**Comment by preetibansalui** [comment]
Hi @alina-jacob, As discussed, the optimal screen reader announcement in this case will be `**1 page number of 11 pages**`.

We cannot use the format `page 1 of 11 pages`, as it would announce the selected page number first, followed by the label.

Please let me know if this works for you.

**Comment by alina-jacob** [comment]
Thanks for the update @preetibansalui, we can remove the word "number" to make the content less verbose. 
So the new expected announcement would be `1, page, of 11 pages`
